### PR TITLE
UOE-9139 - Native LineItem creation for OWSDK

### PR DESCRIPTION
**Pull request by randeepsiddhu**
As a SE, I should be able to use Python line item generation tool to create native ad LI for OWSDK publishers

### CODE REVIEW

**Comment by pm-isha-bharti** [comment]
Hi @randeepsiddhu 

Can you please update the README as well for the new param introduced?

**Review by pm-isha-bharti — CHANGES_REQUESTED** [review]
Hi @randeepsiddhu 

Can you please update the README for the new param introduced?

File: settings.py [hunk 1]
```diff
@@ -157,6 +157,9 @@
 # you can specify an array for multiple creative templates
 OPENWRAP_CREATIVE_TEMPLATE = 'ganeshformat'
 
+# option to set user-defined variable for NATIVE creative type (used only for NATIVE setup type)
```
Comment (pm-isha-bharti) [inline settings.py:160]: It will be used for In-app as well, right ?

**Review by pm-isha-bharti — COMMENTED** [review]
Do we want to introduce new `OPENWRAP_SETUP_TYPE` like `INAPP_NATIVE` like we have for `INAPP_VIDEO` setup? This will be helpful if we have more In-app native changes coming in future.

So instead of adding below check at multiple places in the code: 
`if setup_type == 'IN_APP' and creative_template_ids is not None`
we can simply add: 
if `if setup_type == 'IN_APP_NATIVE'

CC: @pm-preety-narvekar @pm-ganesh-salpure

File: settings.py (hunk 1 above)
Comment (randeepsiddhu) [reply settings.py]: This variable is used only when the creative type is NATIVE, I have fixed the typo.

**Comment by randeepsiddhu** [comment]
- Updated the README
- Fixed the typo in settings file
- Will consider adding setup_type IN_APP_NATIVE in next iteration

File: dfp/create_creatives.py [hunk 2]
```diff
@@ -153,26 +153,37 @@ def create_creative_configs_for_native(advertiser_id,creative_template_ids,
...
           advertiser_id=advertiser_id,
           creative_template_id=template_id,
+          user_def_var=user_def_var
         )
         creative_configs.append(config)
       
   return creative_configs
   
 
-def create_creative_config_native(name, advertiser_id, creative_template_id):
```
Comment (pm-isha-bharti) [inline dfp/create_creatives.py:162]: Can you please include a unit test for this?

**Comment by randeepsiddhu** [comment]
Hi @pm-isha-bharti and @pm-ganesh-salpure,
I have made the changes based on the review comments.
Please take a look and let me know.
Have tested with the following SETUP_TYPE: WEB, IN_APP, IN_APP_VIDEO, NATIVE, IN_APP_NATIVE. 
Refer orders with string 9139 on 15671365 GAM account.
CC: @pm-preety-narvekar

File: README.md [hunk 3]
```diff
@@ -91,7 +91,7 @@ If successful, it returns all the orders in your GAM account.
...
 |`PREBID_BIDDER_CODE`|The value of [`pwtpid`](https://github.com/PubMatic/OpenWrap#wrapper-keys-sent-to-dfp) for this partner. Set to `None` to generate line items for all partners. Use array of strings if the line should match multiple partners.|string or array of strings|
-|`OPENWRAP_SETUP_TYPE`|Which type of creative to use.  Options are `WEB`, `WEB_SAFEFRAME`, `AMP`, `IN_APP`, `IN_APP_VIDEO`, `NATIVE`, `VIDEO`, `JWPLAYER`,`ADPOD`.|string|
+|`OPENWRAP_SETUP_TYPE`|Which type of creative to use.  Options are `WEB`, `WEB_SAFEFRAME`, `AMP`, `IN_APP`, `IN_APP_VIDEO`, `IN_APP_NATIVE`, `NATIVE`, `VIDEO`, `JWPLAYER`,`ADPOD`.|string|
 |`OPENWRAP_BUCKET_CSV`|This CSV lists buckets and price granularity; it sets `pwtpb` targeting for each line item for adpod setup  and  `pwtecp` targeting for each line item for other setups. For Adpod setup set this parameter to one of the `Inline_Header_Bidding_{Price Granularity}.csv file`.| string|
 |`OPENWRAP_CREATIVE_TEMPLATE` |The creative template name for Native Lineitems. This is only required when `OPENWRAP_SETUP_TYPE`=`NATIVE`. | string |
```
Comment (pm-isha-bharti) [inline README.md:96]: Add setup "IN_APP_NATIVE" as well here.

File: settings.py [hunk 4]
```diff
@@ -140,7 +140,7 @@
 #]
 
 # OpenWrap Creative Type
-#  One of "WEB", "WEB_SAFEFRAME", "AMP", "IN_APP", "IN_APP_VIDEO", "NATIVE", "VIDEO", "JWPLAYER", "ADPOD"
+#  One of "WEB", "WEB_SAFEFRAME", "AMP", "IN_APP", "IN_APP_VIDEO", "IN_APP_NATIVE", "NATIVE", "VIDEO", "JWPLAYER", "ADPOD"
```
Comment (pm-isha-bharti) [inline settings.py:143]: Please mention `IN_APP_NATIVE` at all places where "`IN_APP`", "`IN_APP_VIDEO`" in the comments.
Ref Line no: 91, 112, 136, etc

**Comment by pm-isha-bharti** [comment]
LGTM

**Review by pm-preety-narvekar — COMMENTED** [review]
Hi @randeepsiddhu 

Can you hardcode value for `Title` as `pubmatic-ow-signal:%%PATTERN:pwtsid%%` for inapp native in the tool itself? It will avoid the typos in setting up this.

**Pull request merged**